application/comit_node/src/settings/mod.rs

```diff
@@ -5,93 +5,188 @@ use crate::seed::Seed;
 use config::{Config, ConfigError, File};
 use libp2p::Multiaddr;
 use log::LevelFilter;
-use serde::Deserialize;
-use std::{ffi::OsStr, net::IpAddr, path::Path, time::Duration};
-
-#[derive(Clone, Debug, Deserialize)]
+use serde::{Deserialize, Serialize};
+use std::{
+    ffi::OsStr,
+    fs,
+    io::Write,
+    net::{IpAddr, Ipv4Addr},
+    path::{Path, PathBuf},
+    time::Duration,
+};
+use url::Url;
+
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
 pub struct ComitNodeSettings {
     pub comit: Comit,
     pub network: Network,
     pub http_api: HttpSocket,
     pub btsieve: Btsieve,
-    #[serde(with = "self::serde_log", default = "default_log")]
-    pub log_level: LevelFilter,
     pub web_gui: Option<HttpSocket>,
+    #[serde(default = "default_log_levels")]
+    pub log_levels: LogLevels,
+}
+
+impl Default for ComitNodeSettings {
+    fn default() -> Self {
+        let comit_listen = "/ip4/0.0.0.0/tcp/8011".parse().unwrap();
+        let btsieve_url = Url::parse("http://localhost:8181").unwrap();
+        let seed = Seed::new_random().unwrap();
```

Comment:
:wrench: Maybe don't unwrap this and give a good error message by matching the result